include/Core/Metas/QueryOverriders/DateQueryOverrider.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace BulkWP\BulkDelete\Core\Metas\QueryOverriders;
+
+use BulkWP\BulkDelete\Core\Base\BaseQueryOverrider;
+
+/**
+ * Class that encapsulates the logic for handling date format specified by the user.
+ *
+ * @package Bulk Delete
+ *
+ * @author  Sudar
+ *
+ * @since   6.0
+ */
+class DateQueryOverrider extends BaseQueryOverrider {
+	/**
+	 * Delete Options.
+	 *
+	 * @var array
+	 */
+	protected $delete_options;
+	/**
+	 * Date format of meta value that is stored in `wp_post_meta` table..
+	 *
+	 * @var string
+	 */
+	protected $meta_value_date_format;
+
+	/**
+	 * Date format of input value that will be compared with meta value.
+	 *
+	 * @since 1.0
+	 *
+	 * @var string
+	 */
+	protected $input_value_date_format;
+
+	/**
+	 * Setup hooks and load.
+	 *
+	 * @since 1.0
+	 */
+	public function load() {
+		add_action( 'parse_query', array( $this, 'parse_query' ) );
+	}
+
+	/**
+	 * Creates query object after processing date with specified date format.
+	 *
+	 * @param array $delete_options Delete Options.
+	 *
+	 * @return \WP_Query $query Query object.
+	 */
+	public function get_query( $delete_options ) {
+		$query = $this->process_date_fields( $delete_options );
+
+		return $query;
+	}
+
+	public function process_date_fields( $delete_options ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method process_date_fields() has a Cyclomatic Complexity of 11. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/app/BulkWP/bulk-delete/pullRequest?prid=2562677)